taskrunner/host_gateway.go

```diff
@@ -171,43 +156,33 @@ func (agbh *AgentHostGateway) prepRemoteHost(hostObj host.Host, sshOptions []str
 	}
 
 	// third, copy over the correct agent binary to the remote machine
-	scpAgentOutput := newCappedOutputLog()
-	scpAgentCmd := &subprocess.ScpCommand{
-		Id:             fmt.Sprintf("scp%v", rand.Int()),
-		Source:         filepath.Join(agbh.ExecutablesDir, executableSubPath(&hostObj.Distro)),
-		Dest:           hostObj.Distro.WorkDir,
-		Stdout:         scpAgentOutput,
-		Stderr:         scpAgentOutput,
+	curlAgentOutput := newCappedOutputLog()
+	curlAgentCmd := &subprocess.RemoteCommand{
+		Id: fmt.Sprintf("curl-%d", rand.Int()),
+		CmdString: fmt.Sprintf("cd '%s' && curl -LO '%s'",
+			hostObj.Distro.WorkDir,
+			fmt.Sprintf("%s/%s", settings.Ui.Url, hostutil.ExecutableSubPath(&hostObj.Distro))),
```

Comment:
why nested fmt.Sprintf? 

could write a function to generate this string

Reply:
done